src/Symfony/Component/Validator/Mapping/Cache/Psr6Cache.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Mapping\Cache;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Validator\Mapping\ClassMetadata;
+
+/**
+ * PSR-6 adapter.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class Psr6Cache implements CacheInterface
+{
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $cacheItemPool;
+
+    public function __construct(CacheItemPoolInterface $cacheItemPool)
+    {
+        $this->cacheItemPool = $cacheItemPool;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function has($class)
+    {
+        $item = $this->cacheItemPool->getItem($this->escapeClassName($class));
```

Comment:
@dunglas that PSR-6 is correct but in this context here it is prone to race conditions anyway if used incorrectly as read() and has() are seperate methods working independently. So your code to retrieve the cache data first for no reason makes no sense and @dmaicher proposal is valid.